src/screens/nano-contract/NanoContractDetail.js

```diff
@@ -116,8 +121,55 @@ function NanoContractDetail() {
     });
   }
 
-  const loadData = async () => {
+  /**
+   * Validates if the transaction is a valid nano contract creation tx
+   * and if it's confirmed by a block already.
+   *
+   * If everything is ok, calls loadData, otherwise calls itself again
+   * in 5s if it's waiting for a block confirmation
+   *
+   * We need this because the state and history are only loaded when the tx is confirmed
+   */
+  const validateAndLoad = async () => {
```

Comment:
_Comment unrelated to line number_

Should we stop trying at some point?

Reply:
The network might take some minutes to confirm the transaction. I think it's ok to query the full node once every 5s to get this information if the user is still with the screen opened.